### ISSUE TITLE
API for OSPOS (Enhancement)

### ISSUE THREAD

**Issue by dxchen90** [jekkos/opensourcepos#615]
Anyone has develop a API for OSPOS for:
**Product**
-Quantity
-Retail Price
-UPC/EAN
**Customer**
-First Name
-Last Name
-Postcode
-Address 1
-Address 2
-City
-State
-Zip
-Email
-Phone number

This would be make our easy if we integrate this to our marketplace to receive customers details. We no need to always create new customer by filling their information.

And for product part, we can have live quantity update to all the marketplaces.
Please contact me if anyone can help on this. Very much appreciate. Thank you

**Comment by daN4cat** [comment]
I'm afraid I don't understand your question. What do you exactly mean with this API?

Are you talking of a RESTful API to query OSPOS or something else?

**Comment by dxchen90** [comment]
Hi,
I explain the situation for more understanding, brcause i not sure rest or query api.
1) i need to pull the order data from marketplaceData need is customer name, address, phone number, email.This can merge into ospos customer details.
2) i need push stock price, sku, retail price to the marketplaces.
Is this need a query Api?

Get Outlook for iOS




On Thu, Jun 2, 2016 at 9:57 AM -0700, "FrancescoUK" <notifications@github.com> wrote:












I'm afraid I don't understand your question. What do you exactly mean with this API?



Are you talking of a RESTful API to query OSPOS or something else?

**Comment by daN4cat** [comment]
OK, so you are asking OSPOS to interface marketplace to share Customer list and Stock.
I never looked into Marketplace but it all depends if they offer an external interface to interact with their database. If yes it's for sure doable, the only thing for the Stock is how you want to keep things in sync re quantity.
In my case I have interfaced Drupal Commerce with OSPOS database, a Drupal Commerce SQL feed keeps stuff in sync reading the quantity field every hour.

**Comment by dxchen90** [comment]
Yes, they offer api. Their systems generate api key for us to apply in to other fullfiment websites.
In my case, i need to key every customer data manually into ospos, one day around 50 customer need to key in manually for every orders from maketplace. 
Any idea to solve this?
Get Outlook for iOS




On Thu, Jun 2, 2016 at 10:09 AM -0700, "FrancescoUK" <notifications@github.com> wrote:












OK, so you are asking OSPOS to interface marketplace to share Customer list and Stock.

I never looked into Marketplace but it all depends if they offer an external interface to interact with their database. If yes it's for sure doable, the only thing for the Stock is how you want to keep things in sync re quantity.

In my case I have interfaced Drupal Commerce with OSPOS database, a Drupal Commerce SQL feed keeps stuff in sync reading the quantity field every hour.

**Comment by daN4cat** [comment]
Does marketplace offer integration with MailChimp?

I have in mind to create a OSPOS connector to MailChimp and save the customers in MailChimp so when we mail out pre-sale alerts we have the list already in MailChimp.

If Marketplace has integration with MailChimp then when I have implemented OSPOS connector you get all in sync.

**Comment by dxchen90** [comment]
No, marketplace doesnt offer mailchimp integration..Hmm.. The marketplaces already generated api keys, ospos needs to have setting to apply the api keys at backend.. Is this call Json api?


Get Outlook for iOS

    _____________________________

**Comment by daN4cat** [comment]
BTW, which marketplace are you referring to?

**Comment by dxchen90** [comment]
Owh, i based in malaysia.
The local marketplaces here are  Lelong.com.my11street.myLazada.com.my
I have some integration guidelines, just i dont know what to with it.

Get Outlook for iOS




On Fri, Jun 3, 2016 at 10:04 AM -0700, "FrancescoUK" <notifications@github.com> wrote:












BTW, which marketplace are you referring to?

**Comment by arteiro** [comment]
@daN4cat  Any guide to integration with drupal commerce, I would like to ict with my store, ppois use Prestashop and integration with drupal is easy to do the migration?

**Comment by daN4cat** [comment]
I used a SQL feed importer on Drupal Commerce to read from OSPOS database items data and quantity.

Transaction always happen from OSPOS side, so Drupal Commerce SQL feed importer is read only.

I assume that creating a smarter module in Drupal Commerce would be a better idea and allowing fancier use cases, but we are happy with that solution and a sync up every 3h to keep quantity in sync.

**Comment by arteiro** [comment]
@daN4cat The module would be very interesting idea, or a virtual store interface with aggregated image to ospos also know it would be too much, but ideas are ideas.
Today I went to see had drupal commerce in Softaculous in my lodging, but not, however found the menu OSPOS in e-commerce (version 2.4).
Maybe in future we will have all integrated as OSPOS is excellent.
I will locally install drupal commerce to try to use the solution you said.
Thank you

**Comment by alvin101** [comment]
In my case, I need a POS that would run stand alone without internet, which yes I can do with ospos, and I happen to like the erp frontaccounting system. 

So my point is an REST that will update at the next time the net is up.

**Comment by waynebr76** [comment]
I have had a developer develop scripts to work with my OSPOS version and Opencart. It requires the Openstock module in Opencart but I run a cron job every 10 minutes that synchronises the stock between OSPOS and Opencart it is a godsave for me. I am just having the script updated so that it updates my stock on Ebay using Openbay (Opencart Module). It is possible but cost us a bit of money and as they have been developed by another developer I don't think I can publicly share.

**Comment by vishwajith-r** [comment]
I think you need to bill online store orders through the OSPOS using a API?

I also wanted to do the same thing. I have a wordpress website. I wanted to bill automatically those orders through the OSPOS. For that I build a Chrome web browser plugging using ChatGPT support. This plugging catch WooCommerce orders via WooCommerce API. And This plugging floating in the OSPOS screen. When I enter the order number, It runs and input details automatically in the OSPOS billing menu. 

This resolved my issue. And I havent change any in the OSPOS code. Then I can install any update without any issue.
